File: Easy/Array/1684. Count the Number of Consistent Strings/Solution.java
class Solution {
    public int countConsistentStrings(String allowed, String[] words) {
      
        boolean[] allowedChars = new boolean[26];
      
        for(char ch: allowed.toCharArray()){
            allowedChars[ch - 'a'] = true;
        }

        int count = 0;

        for(String word: words){
          
            boolean isConsistent = true;
          
            for(char ch: word.toCharArray()){
              
                if(!allowedChars[ch - 'a']){
                    isConsistent = false;
                    break;
                }
            }
          
            if(isConsistent)
                count++;
        }
      
        return count;
    }
}

File: Easy/Array/1684. Count the Number of Consistent Strings/Solution.kt
class Solution {
    fun countConsistentStrings(allowed: String, words: Array<String>): Int {

        val allowedChars = BooleanArray(26)

        for(ch in allowed) {
            allowedChars[ch - 'a'] = true
        }

        var count = 0

        for(word in words){

            var isConsistent = true

            for(ch in word){
                if(!allowedChars[ch - 'a']){
                    isConsistent = false
                    break
                }
            }

            if(isConsistent)
                count++

        }

        return count
    }
}
